src/dune/dune.py

```diff
@@ -470,6 +470,148 @@ def system_newaccount(self, name, creator=None, pub=None, private=None, addition
+
+    def update_app(self, path: str, dependency_name: str, lang: str,
+                   cmplr_opts: str = None, link_opts: str = None):
+
+        opts: list = []
+        if cmplr_opts:
+            opts.append(cmplr_opts)
+        if link_opts:
+            opts.append(link_opts)
+
+        container_dir = self._docker.abs_host_path(path)
+
+        self._docker.execute_cmd(["antler-proj", "update", container_dir, "app",
+                                  dependency_name, lang])
+
+    def update_lib(self, path: str, dependency_name: str, lang: str,
+                   cmplr_opts: str = None, link_opts: str = None):
+
+        opts: list = []
+        if cmplr_opts:
+            opts.append(cmplr_opts)
+        if link_opts:
+            opts.append(link_opts)
+
+        container_dir = self._docker.abs_host_path(path)
+
+        self._docker.execute_cmd(["antler-proj", "update", container_dir, "lib",
+                                  dependency_name, lang] + opts)
+
+    def update_dep(self, path: str,        # project path
+                   object_name: str,       # object name (app/lib)
+                   dependency_name: str,   # dependency name
+                   location: str = None,   # location of dep
+                   tag_rel: str = None,    # tag/release number
+                   hash_str: str = None):  # hash
+
+        opts: list = []
+        if location:
+            opts.append(location)
+        if tag_rel:
+            opts.append(tag_rel)
+        if hash_str:
+            opts.append(hash_str)
+
+        container_dir = self._docker.abs_host_path(path)
+
+        self._docker.execute_cmd(["antler-proj", "update", container_dir, "dep",
+                                  object_name, dependency_name] + opts)
+
+    def remove_dep(self, path: str,        # project path
+                   object_name: str,       # object name (app/lib)
+                   dependency_name: str):   # dependency name
+
+        container_dir = self._docker.abs_host_path(path)
+        self._docker.execute_cmd(["antler-proj", "remove", container_dir, "dep",
+                                  object_name, dependency_name])
+
+    def remove_app(self, path: str,        # project path
+                   app_name: str):       # app name
+
+        container_dir = self._docker.abs_host_path(path)
+        self._docker.execute_cmd(["antler-proj", "remove", container_dir, "app", app_name])
+
+    def remove_lib(self, path: str,        # project path
+                   lib_name: str):       # lib name
+
+        container_dir = self._docker.abs_host_path(path)
+        self._docker.execute_cmd(["antler-proj", "remove", container_dir, "lib", lib_name])
+
+    def build_project(self, path, clean_build: bool = False):
+        container_dir = self._docker.abs_host_path(path)
+        if clean_build:
+            self._docker.execute_cmd(["antler-proj", "build", container_dir, "--clean"])
```

Comment:
Does you antler-proj handle `--clean`?
I took antler-proj from`main` and it doesn't.
If your antler-proj doesn't handle `--clean` as well can you please create an issue in antler-proj?

Reply:
> Does you antler-proj handle `--clean`? I took antler-proj from`main` and it doesn't. If your antler-proj doesn't handle `--clean` as well can you please create an issue in antler-proj?

No it doesn't bit this feature is required in #124 so we agreed format of the command with @ScottBailey and I made a corresponding issue https://github.com/AntelopeIO/antler-proj/issues/18